server/routes/sources/sources-router.js

```diff
@@ -2,53 +2,84 @@ import express from 'express';
 import AppError from '../../errors/AppError.js';
 import {
   findSourceCountry,
-  updateSourceById,
+  getSourceByIdSourceName,
+  getCrosswalkByIdSourceName,
+  updateSourceByIdSourceName,
   createSource,
   createCrosswalk,
   getAllSources,
+  updateCrosswalkByIdSourceName,
 } from './sources-queries.js';
+import validateSource from './sources-validations.js';
 
 const sourcesRouter = express.Router();
 
 sourcesRouter.get('/', async (req, res) => {
-  const { id, country } = req.query;
+  const { country, idSourceName, sources } = req.query;
 
-  const idSource = !id ? '*' : id;
-  const source = !country
-    ? await getAllSources()
-    : await findSourceCountry(country, idSource);
-  res.status(200).json(source ?? {});
+  if (!idSourceName && sources === 'All') {
+    const foundSources = await getAllSources();
+    if (!foundSources || foundSources.length === 0)
+      throw new AppError(400, 'Error getting source');
+    res.status(200).json(foundSources ?? {});
```

Comment:
Could sources be null? Would that be caught in line 22? If not, do we want to check that in line 22 as well?